File: 10) Java OOP - February2019/07. Encapsulation - Exercises/01-02. Box/figures/Box.java
package figures;

public class Box {

    private double length;

    private double width;

    private double height;

    public Box(double length, double width, double height) throws Exception {
        this.setLength(length);
        this.setWidth(width);
        this.setHeight(height);
    }

    public double calculateSurfaceArea() {
        return (2 * length * width) + (2 * length * height) + (2 * width * height);
    }

    public double calculateLateralSurfaceArea() {
        return (2 * length * height) + (2 * width * height);
    }

    public double calculateVolume() {
        return length * width * height;
    }

    @Override
    public String toString() {
        return String.format(
                "Surface Area - %.2f%nLateral Surface Area - %.2f%nVolume - %.2f",
                calculateSurfaceArea(),
                calculateLateralSurfaceArea(),
                calculateVolume()
        );
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) throws Exception {
        if (length < 0) {
            throw new Exception("Length cannot be zero or negative");
        }
        this.length = length;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) throws Exception {
        if (width < 0) {
            throw new Exception("Width cannot be zero or negative");
        }
        this.width = width;
    }

    public double getHeight() {
        return height;
    }

    public void setHeight(double height) throws Exception {
        if (height < 0) {
            throw new Exception("Height cannot be zero or negative");
        }
        this.height = height;
    }
}

File: Software_University/04_PHP_Web_Development_Basics/12) OOP FUNDAMENTALS - Exercises/02. Box Data Validation/Box.php
<?php

namespace Van\Classes\Second;


class Box
{
    private $length;
    private $width;
    private $height;

    public function __construct(float $length, float $width, float $height)
    {
        $this->setLength($length);
        $this->setWidth($width);
        $this->setHeight($height);
    }

    public function getSurfaceArea(): string
    {
        $surface = (
            (2 * $this->length * $this->height)
            + (2 * $this->length * $this->width)
            + (2 * $this->width * $this->height)
        );

        return number_format(round($surface, 2), 2);
    }

    public function getVolume(): string
    {
        $volume = $this->length * $this->width * $this->height;

        return number_format(round($volume, 2), 2);
    }

    public function getLateralSurfaceArea(): string
    {
        $latteralSurfaceArea = (2 * $this->length * $this->height) + (2 * $this->width * $this->height);

        return number_format(round($latteralSurfaceArea, 2), 2);
    }

    /**
     * @param float $length
     */
    private function setLength(float $length)
    {
        if ($length <= 0) {
            exit('Length cannot be zero or negative.');
        }

        $this->length = $length;
    }

    /**
     * @param float $width
     */
    private function setWidth(float $width)
    {
        if ($width <= 0) {
            exit('Width cannot be zero or negative.');
        }

        $this->width = $width;
    }

    /**
     * @param float $height
     */
    private function setHeight(float $height)
    {
        if ($height <= 0) {
            exit('Height cannot be zero or negative.');
        }

        $this->height = $height;
    }
}
